src/main/java/com/auctionapp/api/service/UserService.java

```diff
@@ -25,30 +36,72 @@ public UserDto getUser(final String email) {
         throw new RuntimeException("User with email " + email + " does not exist!");
 	}
 
+	public UserDto getUserById(final UUID id) {
+		Optional<User> user = userRepository.findById(id);
+        if (user.isPresent()) {
+			return toPayload(user.get());
+		}
+        throw new RuntimeException("User with id " + id + " does not exist!");
+	}
+
+	public void deactivateUser(final UUID id) {
+		Optional<User> user = userRepository.findById(id);
+		user.get().setStatus(Status.INACTIVE);
+		userRepository.save(user.get());
+	}
+
+	public UserDto update(UUID id, UserDto user) {
+        User updatedUser = fromPayload(user);
+        updatedUser = userRepository.save(updatedUser);
+		paymentDetailsRepository.save(updatedUser.getPaymentDetails());
+		shippingDetailsRepository.save(updatedUser.getShippingDetails());
+        return toPayload(updatedUser);
+	}
+
 	public static User fromPayload(final UserDto payload) {
 		User user = new User(payload.getId(),
 							payload.getFirstName(),
 							payload.getLastName(),
 							payload.getEmail(),
 							payload.getPassword(),
+							payload.getPhoneNum(),
+							payload.getGender(),
+							payload.getDateOfBirth(),
 							payload.getCreatedAt(),
 							payload.getUpdatedAt(),
-							payload.getRole()
+							payload.getRole(),
+							payload.getStatus()
 							);
+		if (payload.getPaymentDetails() != null) {
+			user.setPaymentDetails(PaymentDetailsService.fromPayload(payload.getPaymentDetails()));
+		}
+		if (payload.getShippingDetails() != null) {
+			user.setShippingDetails(ShippingDetailsService.fromPayload(payload.getShippingDetails()));
+		}
```

Comment:
There should be a constructor that accepts these and you can check whether they're null or not via a ternary operator

Reply:
Do I make another constructor with these fields and then if they are not null, to call that constructor or check if null in the existing constructor then assign them or something else?

Reply:
You can use the existing constructor. On second thought the null check in the constructor might not even be necessary if these are null by default anyway when they aren't set. Just make sure you don't get an exception when invoking the constructor.

Reply:
I get null exception even before entering user constructor, when I'm calling ShippingDetailsService.fromPayload(payload.getShippingDetails()) method to send this to user constructor, it is NullPointerException when trying to call shipping details constructor in fromPayload method.

I am not sure how I should re-write this without checking for null before calling these methods or checking for null in them before calling constructors

Reply:
Your `fromPayload` methods can use a top level check to see if the entire model is null before accessing any properties. If it is, return null, otherwise return the constructed object.